include/dr/mhp/algorithms/inclusive_exclusive_scan_impl.hpp

```diff
@@ -83,7 +109,10 @@ auto inclusive_scan_impl_(R &&r, O &&d_first, BinaryOp &&binary_op,
     auto [global_in, global_out] = global_seg;
     if (dr::ranges::rank(global_in) == rank) {
       auto local_out = dr::ranges::__detail::local(global_out);
-      auto back = use_sycl ? sycl_get(local_out.back()) : local_out.back();
+      auto local_in = dr::ranges::__detail::local(global_in);
+      auto back = use_sycl ? binary_op(sycl_get(local_out.back()),
```

Comment:
I think this change broke inclusive scan.
Depending on whether you have inclusive or exclusive there should be preious or new behavior.